dpreference/src/main/java/me/dozen/dpreference/PreferenceProvider.java

```diff
@@ -114,12 +122,11 @@ public int delete(Uri uri, String selection, String[] selectionArgs) {
     }
 
     @Override
-    public int update(Uri uri, ContentValues values, String selection, String[] selectionArgs) {
+    public int update(Uri uri, ValuesBucket values, DataAbilityPredicates predicates) {
         PrefModel model = getPrefModelByUri(uri);
```

Comment:
model variable could be null as getPrefModelByUri throws IllegalArgumentException in some scenarios. Please do a null check for model variable and surround it with try/catch. Do the same in all the places where getPrefModelByUri(uri) is called

Reply:
Done